Week3/homework/step2-3.js

```diff
@@ -42,17 +30,11 @@ console.log('while', repeatStringNumTimesWithWhile('abc', 3));
 function repeatStringNumTimesWithDoWhile(str, num) {
   // eslint-disable-next-line
   let result = '';
-
-  if (num <= 0) {
-    result = '';
-  } else {
-    let i = 0;
-    do {
-      result += str;
-      i++;
-    } while (i < num);
-  }
-
+  let i = 0;
+  do {
```

Comment:
As the loop condition for the `do-while` loop is tested at the end, the loop body will always be executed at least once. In this case therefore you do need an `if` statement to ensure that the loop body is not executed if `num <= 0`. This illustrates why a `do-while` loop is pretty useless in practice. And it can always be rewritten using a (safe) regular `while` loop.

Reply:
I see now dear Jim sorry for late understand.